feat(android): Add contentSize to scroll event

m1ga opened the pull request:
- Parity for `contentSize` inside the `scroll` event of a `ScrollView`

##### TEST CASE
```js
const win = Ti.UI.createWindow({ backgroundColor: 'gray' });
const view = Ti.UI.createView({
    backgroundColor: 'red',
    top: 0,
    width: 20,
    height: 1020
});
const scrollView = Ti.UI.createScrollView({
    height: Ti.UI.FILL,
    width: Ti.UI.FILL
});

scrollView.addEventListener('scroll', e => {
	console.log('contentSize:', e.contentSize);
});

win.add(scrollView);
scrollView.add(view);
win.open();
```
```
contentSize: {"width":411,"height":1020}
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-27283)

ewanharris committed:
fix(ios): fallback to non-widget require in catch

This ensures that when the alloy require fails we try the normal require

Fixes TIMOB-27195

m1ga committed:
fix(android): add contentSize to scroll event

build commented:
<!--
  1 failure:  :microscope: Ther...
  0 warning: 
  3 messages
  
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :microscope: There are library changes, but no changes to the unit tests. That's OK as long as you're refactoring existing code, but will require an admin to merge this PR. Please see <a href="https://github.com/m1ga/titanium_mobile/blob/contentSize/README.md#unit-tests">README.md#unit-tests</a> for docs on unit testing.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:tada: Another contribution from our awesome community member, m1ga! Thanks again for helping us make Titanium SDK better. :thumbsup:</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: All tests are passing
Nice one! All 1930 tests are passing.
(There are 226 tests skipped)
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 42055fa2cae937925a35a0211b811aced2d304f4
</p>

jquick-axway commented on android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java:
```diff
@@ -448,9 +445,26 @@ protected void onScrollChanged(int l, int t, int oldl, int oldt)
+				TiDimension dimH = new TiDimension(child.getHeight(), TiDimension.TYPE_HEIGHT);
+				contentWidth = TiConvert.toInt(dimW.getAsDefault(child), 0);
+				contentHeight = TiConvert.toInt(dimH.getAsDefault(child), 0);
+			}
```
Variables `contentWidth` and `contentHeight` should be of type `double` and not `int`. And we should not convert the `double` returned by `getAsDefault()` to an integer.

The reason is because the `contentSize` you are providing needs to use the units defined in the "tiapp.xml", which will typically be "dp" by default (not pixels). "dp" based values can be **fractional**. Especially on an "hdpi" Android device which has a `1.5` scaling factor. Test it out in an "hdpi" emulator and you'll see what I mean.

m1ga replied:
done

jquick-axway commented on android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java:
```diff
@@ -575,13 +589,28 @@ protected void onScrollChanged(int l, int t, int oldl, int oldt)
+				TiDimension dimH = new TiDimension(child.getHeight(), TiDimension.TYPE_HEIGHT);
+				contentWidth = TiConvert.toInt(dimW.getAsDefault(child));
+				contentHeight = TiConvert.toInt(dimH.getAsDefault(child));
+			}
```
Same as above. Content width/height should be doubles, not ints.

m1ga replied:
done

jquick-axway commented on apidoc/Titanium/UI/ScrollView.yml:
```diff
@@ -114,9 +114,9 @@ events:
         type: Dictionary
-        platforms: [iphone, ipad]
+        platforms: [android, iphone, ipad]
+        since: { iphone: "5.2.0", ipad: "5.2.0", android: "8.2.0" }
```
Titanium 8.2.0 is planned to be an iOS 13 only release.
No Android or Windows changes are allowed.

Would you mind changing the version from 8.2.0 to 8.3.0 please?

jquick-axway commented:
Thanks for the contribution @m1ga .
I wasn't aware of the event's "contentSize" property on iOS. I'm sure this will help.

m1ga commented:
Thanks, all issues fixed

garymathews commented on android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java:
```diff
@@ -448,9 +445,26 @@ protected void onScrollChanged(int l, int t, int oldl, int oldt)
 				getProxy().fireEvent(TiC.EVENT_DRAGSTART, data);
 			}
+
+			double contentWidth = 0;
```
Maybe make a method that returns a `contentData` `KrollDict` so we don't have duplicate code?

m1ga replied:
good idea! Will change that

m1ga commented on android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java:
```diff
@@ -1061,6 +1063,27 @@ public void scrollToTop()
+		TiScrollViewLayout lay = getLayout();
+
+		if (lay.getChildCount() > 0) {
+			final View child = lay.getChildAt(0);
```
I need to check that again. I think it might return the wrong child

jquick-axway committed:
fix(android): splash open animation sometimes stutters on cold start (#11141)

Fixed bug where opening a window while splash is in mid-transition causes splash to adopt child window's transition. Causes noticeable stutter.
  * Especially an issue as of Android Q beta 4.
  * Modified so that activity windows launched from splash override pending transition with fade-in transition instead. (Looks better than slide-up animation anyways.)

Fixes TIMOB-27328

jquick-axway committed:
fix(android): improve merge of <uses-feature/> elements (#11146)

- Fixed so that Android setting <uses-feature/> required "true" in "tiapp.xml" should override "false" setting in libraries.
  * Merge is supposed to do a logical-OR on the "android:required" value.
- Added "tools:replace" attribute support to <uses-feature/>.
  * Allows overriding "android:required" to "false" if set "true" by library. (Only means of doing this.)

Fixes TIMOB-27304

ewanharris committed:
fix(android): fix asset encryption on windows 32bit java (#11159)

Fixes TIMOB-27266

jquick-axway committed:
fix(android): ProgressIndicator dialog handling (#11143)

* [TIMOB-27104] ProgressIndicator logs "WindowLeaked" exception when hiding dialog and closing window at same time as of 8.0.2
* [TIMOB-27308] Determinant ProgressIndicator ignores "value" property before shown
* [TIMOB-27309] Cannot re-show ProgressIndicator dialog if auto-closed by previous window
* Removed dialog's handleMessage() related code. No longer needed since JS runs on main UI thread.
* added unit test for Ti.UI.Android.ProgressIndicator

Fixes TIMOB-27104, TIMOB-27308, TIMOB-27309

vijaysingh-axway committed:
fix(ios): watchsession module loaded (#11182)

Fixes TIMOB-27343

janvennemann committed:
fix(ios): use device build as base for universal module framework (#11185)

Fixes TIMOB-27279

vijaysingh-axway committed:
fix(ios): remove invalid architecture from framework for adhoc build (#11183)

Fixes TIMOB-27261

vijaysingh-axway committed:
fix(ios): console and tiapi multiple parameter log (#11164)

Fixes TIMOB-27342

garymathews committed:
chore(8_3_X): update aps-analytics (#11180)

* chore: update aps-analytics

* fix(ios): improve uncaught exception stack traces

* build(lint): fix formatting

janvennemann committed:
feat(ios): expose clearMotionEffects api (#11181)

Fixes TIMOB-26566

garymathews reviewed (approved):
Updated PR, LGTM!

No tests as this requires user input.

hansemannn committed:
fix(ios): remove old UIWebView left overs (#11191)

keerthi1032 commented:
FR passed.shows content size inside scroll  view
Test Environment:
Name                        = Mac OS X
  Version                     = 10.14.5
  Architecture                = 64bit
 Node.js
  Node.js Version             = 10.16.2
  npm Version                 = 6.9.0
Titanium CLI
  CLI Version                 = 5.2.1
Titanium SDK
  SDK Version                 = local sdk 8.3.0.v20190903144732
Device -samsung s5 android 6, oneplus 5t android 9
Simulator-pixel android 9,nexus 6p android 7

keerthi1032 committed:
Merge branch '8_3_X' into contentSize

sgtcoolguy commented:
manually rebased on top of master and merged.

build commented on tests/Resources/ti.ui.android.progressindicator.addontest.js:
```diff
@@ -0,0 +1,211 @@
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Titanium */
```
<!--
  1 failure:  tests/Resources/t...
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.ui.android.progressindicator.addontest.js;
  Line: 8;
-->
- :no_entry_sign: tests/Resources/ti.ui.android.progressindicator.addontest.js line 8 – 'Titanium' is already defined as a built-in global variable. (no-redeclare)
- :warning: tests/Resources/ti.ui.android.progressindicator.addontest.js line 8 – 'Titanium' is defined but never used. (no-unused-vars)
